internals/metrics_models.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import ndb
+
+
+# UMA metrics.
+class StableInstance(ndb.Model):
+  created = ndb.DateTimeProperty(auto_now_add=True)
+  updated = ndb.DateTimeProperty(auto_now=True)
+
+  property_name = ndb.StringProperty(required=True)
+  bucket_id = ndb.IntegerProperty(required=True)
+  date = ndb.DateProperty(verbose_name='When the data was fetched',
+                         required=True)
+  day_percentage = ndb.FloatProperty()
+  rolling_percentage = ndb.FloatProperty()
+
+
+class AnimatedProperty(StableInstance):
+  pass
+
+
+class FeatureObserver(StableInstance):
+  pass
```

Comment:
What is the purpose of these empty inheriting classes? Used just for naming clarity?

Reply:
Each ndb.Model subclass creates a different entity kind in datastore.  It is like having separate SQL tables.  This code was here before me, but I think the purpose is just to keep separate types of metrics data in separate entities so that queries done per-entity would retrieve only those entities.